// @chakra-ui/cli/dist/cjs/utils/init-cli.cjs
'use strict';

var welcome = require('cli-welcome');
var fs = require('node:fs');

var _documentCurrentScript = typeof document !== 'undefined' ? document.currentScript : null;
function readPackageJson() {
  let url;
  try {
    url = new URL("../../../package.json", (typeof document === 'undefined' ? require('u' + 'rl').pathToFileURL(__filename).href : (_documentCurrentScript && _documentCurrentScript.src || new URL('utils/init-cli.cjs', document.baseURI).href)));
  } catch {
    url = new URL("../../package.json", (typeof document === 'undefined' ? require('u' + 'rl').pathToFileURL(__filename).href : (_documentCurrentScript && _documentCurrentScript.src || new URL('utils/init-cli.cjs', document.baseURI).href)));
  }
  return JSON.parse(fs.readFileSync(url, "utf-8"));
}
async function initCLI() {
  const pkgJSON = readPackageJson();
  const { default: updateNotifier } = await import('update-notifier');
  welcome({
    title: "Chakra UI CLI",
    tagLine: `by Chakra UI`,
    bgColor: `#319795`,
    color: `#FFFFFF`,
    bold: true,
    clear: false,
    version: pkgJSON.version
  });
  updateNotifier({
    pkg: pkgJSON,
    shouldNotifyInNpmScript: true,
    updateCheckInterval: 1e3 * 60 * 60 * 24 * 3
    // 3 days
  }).notify({ isGlobal: true, message: "New version of CLI available" });
}

exports.initCLI = initCLI;
